Add guidelines for accepting component level contributions (#185)
buildpacks/community: contributors/guide.md

Contribution to this project goes beyond the notion of code contributions. We value any help that you may be able to provide.

#### Component

> Components developed outside of the Cloud Native Buildpacks project can be accepted following this process:

Reviewer: Feedback:

1. This is a large wall of text for a type of contribution that is bound to happen less frequently.
2. The contents are pretty much verbaidum to that of the RFC.

Given that this document is more intended to new contributors would it be easier on the eyes if we simply had a link to the RFC. For example:

```suggestion
Components developed outside of the Cloud Native Buildpacks project can be accepted following the process defined in [this RFC](https://github.com/buildpacks/rfcs/blob/main/text/0086-component-level-contrib.md#what-it-is).
```

- Component-level contributions must be proposed as project-wide RFCs
- Proposed contributions must primarily support the goals and use cases of the project and may not contain unrelated functionality that would not otherwise be accepted via the [RFC process](https://github.com/buildpacks/rfcs).
  - Qualifying examples:
    1. A Github Action that performs buildpack builds
    1. A Concourse resource that creates builder images
    1. A v2 Heroku buildpack that runs Cloud Native Buildpack (v3) builds on platforms that only support older v2 buildpacks.
  - Non-qualifying examples:
    1. A generic CI/CD tool that includes functionality to run buildpack builds, but primarily supports use cases outside of the project (e.g., Tekton)

> - If a proposed contribution provides functionality that is already provided by the project, such functionality must be provided for use in a non-overlapping context, or the proposal must include a plan for consolidation.

Reviewer 2: How strongly do we feel about this? Should it be "should" instead of "must"?

> - If a proposed contribution provides functionality that is already provided by the project, such functionality must be provided for use in a non-overlapping context, or the proposal must include a plan for consolidation.

Reviewer 3: ```suggestion
- If a proposed contribution provides functionality that is already provided by the project, such functionality must be provided for use in a non-overlapping context, or the proposal should include a plan for consolidation.
```

  - Qualifying examples:
    1. Buildpack language bindings for a language other than those already supported by the project (such as Go)
    1. An alternative implementation of the buildpack lifecycle launcher in Rust, with a documented plan to deprecate the original Go version
  - Non-qualifying examples:
    1. An alternative implementation of the lifecycle in Rust that would exist concurrently with the Go version in perpetuity

> - If a proposed contribution is an adapter or integration with another service or technology, that technology must be notable and/or widely used (at the Core Team’s discretion).
> - If a component is proposed for donation by one or more Core Team members individually, those Core Team members must abstain from voting over its acceptance.
> - If a component is proposed for donation by one or more Core Team members’ employer or employers, those Core Team members must abstain from voting over its acceptance. Supermajority consensus of non-abstaining Core Team members is still required.
> - Component-level contributions are subject legal review by the Cloud Native Computing Foundation / Linux Foundation. Adherance to these guidances does not guarantee acceptance. The investigation of this is to be done by the Core Team before acceptance.

Reviewer 3: ```suggestion
- If a proposed contribution is an adapter or integration with another service or technology, that technology must be notable and/or widely used (at the TOC’s discretion).
- If a component is proposed for donation by one or more TOC members individually, those TOC members must abstain from voting over its acceptance.
- If a component is proposed for donation by one or more TOC members’ employer or employers, those TOC members must abstain from voting over its acceptance. Supermajority consensus of non-abstaining TOC members is still required.
- Component-level contributions are subject legal review by the Cloud Native Computing Foundation / Linux Foundation. Adherance to these guidances does not guarantee acceptance. The investigation of this is to be done by the TOC before acceptance.
```
